dags/hindawi/utils.py

```diff
@@ -0,0 +1,46 @@
+import uuid
+from datetime import date, datetime
+from io import BytesIO
+from xml.etree import ElementTree
+
+from airflow.api.common import trigger_dag
+
+
+def save_file_in_s3(data, repo):
+    if bool(data):
+        date_today = date.today().strftime("%Y-%m-%d")
+        prefix = datetime.now().strftime("%Y-%m-%dT%H:%M")
```

Comment:
since you have a date in `date_today`, do you need a date also in prefix?

Reply:
yes, because the same day we can have more than one file downloaded. Let's say we downloaded 5 files today.
So, the "folder" in S3 will be the date "2022-08-10", however, the key of the file itself shows also the time when it was downloaded. In the end,  structure looks like this:
2022-08-10/2022-08-10T17:33.xml
2022-08-10/2022-08-10T18:36.xml
2022-08-10/2022-08-10T19:00.xml
2022-08-10/2022-08-10T21:01.xml
2022-08-10/2022-08-10T23:33.xml